_posts/2021-03-31-relaxing-requirements-of-moved-from-objects.md

```diff
@@ -139,24 +152,29 @@ for (std::string line; std::getline(std::cin, line);) {
 
 However, the concern over changing the documentation is one that should be considered in light of weakening the requirements retroactively as a defect. As that would mean even citing a specific version of the standard would break. 
 
+<!-- I don't really understand the last sentence, and my sense is that politically, it's better not to try to pull off this line of argumentation, and just make a best effort at assessing the risk.  I think there are some requirements that we absolutely wouldn't consider weakening because the risk is too great.  In fact I am starting to think the better line here is to argue that Geoffrey misinterpreted the standard wording and the non-normative note is wrong, so this doesn't represent an actual weakening of requirements… IIUC  -->
+
+
 ## Technical Specifications
 
 We need a general requirement regarding _the domain of an operation_. Borrowing from [the text for input iterators](http://eel.is/c++draft/iterator.cpp17#input.iterators-2):
 
 > Unless otherwise specified, there is a general precondition for all operations that the requirements hold for values within the _domain of the operation_.
 {: .proposed}
 
-> The term _domain of the operation_ is used in the ordinary mathematical sense to denote the set of values over which an operation is (required to be) defined. This set can change over time. Each component may place additional requirements on the domain of an operation. These requirements can be inferred from the uses that a component makes of the operation and is generally constrained to those values accessible through the operation's arguments.
+> The term _domain of the operation_ is used in the ordinary mathematical sense to denote the set of values over which an operation is (required to be) defined. This set can change over time. Each component may place additional requirements on the domain of an operation. <!-- Is this how we deal with, e.g., NaN, which *is* in the domain of ==? -->These requirements can be inferred from the uses that a component makes of the operation and is generally constrained to those values accessible through the operation's arguments.
```

Comment:
`NaN` is in the domain of `operator==()` but not in the domain of EqualityComparable. Right now, the standard is broken with regards to NaN. It is both mine and Howard's opinion that with the current wording you cannot sort an array of floating-point values with operator<() even if the range doesn't contain a NaN

Reply:
I think you mean, “yes.” 😉  EqualityComparable is “additional requirements on the domain” AFAICT.